Use make moby image

aanm opened the pull request:
- Fixed `make docker-image` broke by past commits. 
- Replaced `docker build` command in k8s Vagrant file tests with `make docker-image`
- Fixed issues resulting on replacing the `docker build` command

aanm committed:
tests: changed docker build to make docker-image

By using Dockerfile in root's directory, docker created a huge build
context since it copies all files, including .git, VMs, etc.

This commit uses the `make docker-image` command since it was designed
to copy all relevant files to a stage directory and build the docker
image from that directory, avoiding the cost of creating huge docker
build contexts.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
Moved make clean from docker/Makefile to Dockerfile

Since `make docker-image` can be used on a system without go installed,
the make clean target was moved to the Dockerfile itself where go is
installed.

Fixes: 3380ba98b9 ("tests: changed docker build to make docker-image")

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
Vagrantfiles: excluding packaging docker/stage for rsync

By adding the replacing the docker image build command with `make
docker-image` in commit 3380ba98b9 it caused concurrent issues with
rsync on the sambe base path among multiple VMs printing messages such
as [1]. Thus this commit excludes the faulty directory:
`contrib/packaging/docker/stage`.

[1]
```
Error: file has vanished: "/root/jenkins/workspace/cilium_cilium_PR-
980-AMWMANE5I6SHMDMUG3VFIE2B2UTSP3OHZOSEKQPS7WG6244I7MSQ/contrib/
packaging/docker/cilium-0.9.90/vendor/gopkg.in/yaml.v2/LICENSE.libyaml"
```

Signed-off-by: André Martins <andre@cilium.io>

tgraf commented on contrib/packaging/docker/Makefile:
```diff
@@ -8,7 +8,7 @@ build: clean
 	find $(BUILDDIR) -name ".*" -prune -exec rm -rf {} \;
 	$(MAKE) -C $(BUILDDIR) clean
-	docker build -t "cilium:cilium-ubuntu-16-04" $(BUILDDIR)
+	docker build -t "cilium:local_build" $(BUILDDIR)
```
Let's make this an environment variable which defaults to `"latest"` instead. The relevant build script can then set that to `"local_build"`. This way the Makefile can be used outside of the automated build system as well.

aanm committed:
packaging: add .git to the list of files to copy

By adding the `git ls-files` command, the local docker image build
depends on the .git directory and it can no longer be ignored.

Fixes: 93922ac5b8 ("daemon: only use tracked files in go-bindata")

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
packaging: add scripts directory to the list of files to copy

Since contrib/scripts/bindata.sh was added to compile the go-bindata.go
file, this directory needs to be added to the list of files to copy.
Without it this directory, running `make docker-image` errors out.

Fixes: 2566872676 ("contrib: add script for checking bindata")

Signed-off-by: André Martins <andre@cilium.io>

tgraf reviewed (approved):
Awesome work! 💯
